.github/workflows/main.yml

```diff
@@ -63,6 +63,49 @@ jobs:
         cd build
         ctest --extra-verbose
 
+  gtest-cpu-sycl:
+    name: Test Google C++ unittest (CPU SYCL)
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791 # v2.5.0
+      with:
+        submodules: 'true'
+    - name: Install SYCL runtime
```

Comment:
there are 2 comments for longterm:

1. move this to scripts as done for oneDAL - https://github.com/oneapi-src/oneDAL/blob/master/.ci/env/apt.sh
2. As i see below, xgboost using conda for env - environment-file: tests/ci_build/conda_env/cpp_test.yml
we have SYCL compiler available in conda as well and this might be a better option to work with it. Also yml files would allow updates with automated tools such as renovate/dependabot